### PR TITLE
Disallow remappings with empty prefix.

### LINKED ISSUE #4405: Remappings might be buggy
The following setup does not work, although it should (at it might have in older versions):

```
/tmp/bla/a.sol:
import "./b.sol";
----
/tmp/bla/b.sol:
contract c {}
----
solc =/tmp/x/ /tmp/bla/a.sol
```

### PR DESCRIPTION
**Pull request by ekpyron**
Fixes #4405.

### CODE REVIEW

**Comment by ekpyron** [comment]
Are there any tests for the standard json compiler? I haven't found any - maybe we should add some (I would use a separate issue/PR for that, though).

File: docs/using-the-compiler.rst [hunk 1]
```diff
@@ -23,14 +23,15 @@ it is also possible to provide path redirects using ``prefix=path`` in the follo
...
 
 This essentially instructs the compiler to search for anything starting with
-``github.com/ethereum/dapp-bin/`` under ``/usr/local/lib/dapp-bin`` and if it does not
-find the file there, it will look at ``/usr/local/lib/fallback`` (the empty prefix
-always matches). ``solc`` will not read files from the filesystem that lie outside of
+``github.com/ethereum/dapp-bin/`` under ``/usr/local/lib/dapp-bin``.
+``solc`` will not read files from the filesystem that lie outside of
 the remapping targets and outside of the directories where explicitly specified source
-files reside, so things like ``import "/etc/passwd";`` only work if you add ``=/`` as a remapping.
+files reside, so things like ``import "/etc/passwd";`` only work if you add ``/=/`` as a remapping.
```
Comment (chriseth) [inline docs/using-the-compiler.rst:32]: Does `/=/` work?

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -58,22 +58,26 @@ using namespace std;
...
 
-void CompilerStack::setRemappings(vector<string> const& _remappings)
+bool CompilerStack::parseRemapping(string const& _remapping, Remapping& _output)
+{
+	auto eq = find(_remapping.begin(), _remapping.end(), '=');
+	if (eq == _remapping.end())
+		return false;
+
+	auto colon = find(_remapping.begin(), eq, ':');
+	if (eq == _remapping.begin() || eq - 1 == colon)
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp]: The purpose of this condition might be clearer if it is done after the assignment.

File: libsolidity/interface/CompilerStack.h [hunk 3]
```diff
@@ -103,8 +110,11 @@ class CompilerStack: boost::noncopyable
 	/// All settings, with the exception of remappings, are reset.
 	void reset(bool _keepSources = false);
 
-	/// Sets path remappings in the format "context:prefix=target"
-	void setRemappings(std::vector<std::string> const& _remappings);
+	// Parses a remapping of the format "context:prefix=target".
+	static bool parseRemapping(std::string const& _remapping, Remapping& _output);
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.h]: Please use `boost::optional<Remapping>`

File: docs/using-the-compiler.rst (hunk 1 above)
Comment (ekpyron) [reply docs/using-the-compiler.rst:32]: Yes, I checked.

**Comment by chriseth** [comment]
I think `layout-of-source-files.rst` also needs some changes.

**Review by chriseth — APPROVED** [review]

**Pull request merged**